libs/community/langchain_community/callbacks/uptrain_callback.py

```diff
@@ -0,0 +1,390 @@
+        self.multi_query_daugher_run_id: str = ""
+
+class UpTrainCallbackHandler(AsyncCallbackHandler):
+    """Callback Handler that logs to uptrain.
+
+    Args:
+        BaseCallbackHandler (_type_): _description_
+    """
+
+    REPO_URL: str = "https://github.com/uptrain-ai/uptrain"
+    ISSUES_URL: str = f"{REPO_URL}/issues"
+    DOCS_URL: str = "https://docs.uptrain.ai"
+
+    import os
+    def __init__(
+            self,
+            project_name_prefix: str = "langchain",
+            key_type = "openai",
+            api_key = os.getenv("OPENAI_API_KEY"),
+            ) -> None:
+        """Initializes the `UpTrainCallbackHandler`."""
+        super().__init__()
+
+        uptrain = import_uptrain()
+
+        import os
+
+        # Set uptrain variables
+        self.schema = UpTrainDataSchema(project_name_prefix=project_name_prefix)
+        self.first_score_printed_flag = False
+
+
+        # Based on whether the user enters an UpTrain API key or an OpenAI API key, the client is initialized
+        # If both are entered, the UpTrain API key is used
+        if key_type == "uptrain":
+            settings = Settings(uptrain_access_token=api_key)
+            self.uptrain_client = APIClient(settings=settings)
+        elif key_type == "openai":
+            settings = Settings(openai_api_key=api_key, evaluate_locally=False)
+            self.uptrain_client = EvalLLM(settings=settings)
+        else:
+            raise ValueError("Invalid key type: Must be 'uptrain' or 'openai'")
+
+            
+        
+    def uptrain_evaluate(
+        self,
+        project_name: str,
+        data: List[Dict[str, str]],
+        checks: List[str],
+    ) -> None:
+        """Run an evaluation on the UpTrain server using UpTrain client."""
+        if self.uptrain_client.__class__.__name__ == "APIClient":
+            uptrain_result = self.uptrain_client.log_and_evaluate(
+                project_name=project_name,
+                data=data,
+                checks=checks,
+            )
+        else:
+            uptrain_result = self.uptrain_client.evaluate(
+                data=data,
+                checks=checks,
+            )
+        self.schema.uptrain_results[project_name].append(uptrain_result)
+
+        score_name_map = {
+            "score_context_relevance": "Context Relevance Score",
+            "score_factual_accuracy": "Factual Accuracy Score",
+            "score_response_completeness": "Response Completeness Score",
+            "score_sub_query_completeness": "Sub Query Completeness Score",
+            "score_context_reranking": "Context Reranking Score",
+            "score_context_conciseness": "Context Conciseness Score",
+            "score_multi_query_accuracy": "Multi Query Accuracy Score",
+        }
+
+        for row in uptrain_result:
+            columns = list(row.keys())
+            for column in columns:
+                if column == "question":
+                    print(f"\nQuestion: {row[column]}")
```

Comment:
Why are there print statements here?

Reply:
The intention is that when a user uses the uptrain callback handler and calls a retriever, information is shown in the following format without the user needing to do any additional setup:

<img width="1383" alt="image" src="https://github.com/langchain-ai/langchain/assets/43818888/5936f07e-a980-4395-974e-69a87312e846">


Do you recommend I use logger or another alternative instead?